File: 卡码/代码随想录/Go/哈希/快乐数/LeetCode_0202.go
//202. 快乐数
package main

func getSum(n int) int{
	sum:=0
	for n>0 {
		sum+=(n%10)*(n%10)
		n/=10
	}
	return sum
}

func isHappy(n int) bool {
	set:=make(map[int]bool)
	for n!=1&&!set[n] {
		n,set[n]=getSum(n),true
	}
	return n==1
}

File: 卡码/代码随想录/1/03-哈希表/Java/src/_03_快乐数/LeetCode_0202.java
//202. 快乐数
package _03_快乐数;

import java.util.HashSet;

public class LeetCode_0202 {
	
}
class Solution {
  public boolean isHappy(int n) {
  	HashSet<Integer> set = new HashSet<Integer>();
  	//如果n不为1且set中不包含n
  	while(n!=1&&!set.contains(n)) {
  		set.add(n);
  		n=getSum(n);
  	}
  	return n==1;
  }
  public int getSum(int n) {
  	int sum=0;
  	while(n>0) {
  		sum+=(n%10)*(n%10);
  		n/=10;
  	}
  	return sum;
  }
}